tests/vibe.web.rest.1140/source/app.d

```diff
@@ -0,0 +1,52 @@
+import std.datetime;
+import vibe.d;
+
+interface IOrientDBRoot
+{
+	@property IOrientDBQuery query();
+}
+
+interface IOrientDBQuery
+{
+	@method(HTTPMethod.GET)
+	@path(":db_name/sql/:query/:result_set_size")
+	Json sql(string _db_name, string _query, int _result_set_size);
+}
+
+class OrientDBRoot : IOrientDBRoot {
+	private OrientDBQuery m_query;
+	override @property IOrientDBQuery query() { return m_query; }
+	public this() { this.m_query = new OrientDBQuery(); }
+}
+
+class OrientDBQuery : IOrientDBQuery {
+	override Json sql(string _db_name, string _query, int _result_set_size) {
+		assert(_db_name == Param1, _db_name);
+		assert(_query == Param2, _query);
+		assert(_result_set_size == Param3, to!string(_result_set_size));
+		return Json.emptyObject;
+	}
+}
+
+enum Param1 = "twitter_data";
+enum Param2 = "select DownloadedDateTime from Message";
```

Comment:
And strange moment:
`URL` stores path string in the `string m_pathString;` field and it contains the decoded path. When we call `localURI`, `m_pathString` converted to `Path` with decoding: `Path(urlDecode(m_pathString))`, and after that `localURI` takes `path.toString`, and calls `filterURLEncode` for it.
Note: when we constructs `URL` from Path, `urlEncode` [is not called](https://github.com/rejectedsoftware/vibe.d/blob/master/source/vibe/inet/url.d#L42) for it, but property `path(Path p)` [does it](https://github.com/rejectedsoftware/vibe.d/blob/master/source/vibe/inet/url.d#L128).
Thus, `m_pathString` may contain encoded or non-encoded string.
I suggest the next idea:
1. `m_pathString` should _always_ contain encoded string.
2. When it constructed from `Path`, string _should_ be forcibly encoded.
3. When it constructed from `string` (via `this(string url_string)` or `@property void localURI(string str)`), it _shouldn't_ be forcibly encoded, but correct encoding should be checked (`checkUrlEncoding(m_pathString)`).
4. `@property string localURI()` should use `m_pathString` as is.
5. REST handler should encode path string independently and use `@property void localURI(string str)` for setting it.